apps/pipelines/tests/test_code_node.py

```diff
@@ -308,3 +308,24 @@ def main(input, **kwargs):
     )
     assert create_runnable(pipeline, nodes).invoke(state)["messages"][-1] == "content from file"
     File.objects.get(id=file.id)
+
+
+@django_db_with_data(available_apps=("apps.service_providers",))
+@mock.patch("apps.pipelines.nodes.base.PipelineNode.logger", mock.Mock())
+def test_render_template_with_context_keys(pipeline, experiment_session):
+    nodes = [
+        start_node(),
+        render_template_node(
+            "input: {{input}}, temp_state.my_key: {{temp_state.my_key}}, "
+            "participant_id: {{participant_details.identifier}}"
```

Comment:
This should fix the test:
```suggestion
            "participant_id: {{participant_data.identifier}}"
```

Reply:
@snopoke  I think it worked , let me know if there are any other suggestions on the update

Reply:
there were conflicting changes in the main tried rebasing it but it got messier so will create a new branch if the conflicts persist.